golem-worker-executor-base/tests/api.rs

```diff
@@ -1887,32 +1907,42 @@ async fn long_running_poll_loop_interrupting_and_resuming_by_second_invocation(
         )
         .await;
 
-    let executor_clone = executor.clone();
-    let worker_id_clone = worker_id.clone();
-    let fiber = tokio::spawn(async move {
-        // Invoke blocks until the invocation starts
-        executor_clone
-            .invoke(
-                &worker_id_clone,
-                "golem:it/api.{start-polling}",
-                vec!["second".into_value_and_type()],
-            )
-            .await
-            .unwrap();
-    });
+    executor
+        .invoke(
+            &worker_id,
+            "golem:it/api.{start-polling}",
+            vec!["second".into_value_and_type()],
+        )
+        .await
+        .unwrap();
 
     executor
         .wait_for_status(&worker_id, WorkerStatus::Running, Duration::from_secs(20))
         .await;
 
+    let mut rx = executor.capture_output_with_termination(&worker_id).await;
+
     {
         let mut response = response.lock().unwrap();
         *response = "first".to_string();
     }
 
-    fiber.await.unwrap();
```

Comment:
This is completed when the second invocation is enqueued. If we had bad luck the first invocation was not completed at that time, causing the test to timeout. 

Looking for the lock message of the first poll loop being completed should be more reliable and work without sleeps

Reply:
Ah this whole test must have been written _before_ we had persistent invocation queues, based on the "// Invoke blocks until the invocation starts" comment. Invoke no longer does this - previously it was blocking on a mutex until the execution actually started.

I've reviewed these tests many times and my eyes always skipped this comment